### PR TITLE
Pre rc10 fixes

### PR DESCRIPTION
**Pull request by smallhive**
Small pre-RC10 fixes

### CODE REVIEW

File: user/signer.go [hunk 1]
```diff
@@ -48,3 +48,40 @@ func NewSignerRFC6979WithID(pk ecdsa.PrivateKey, id ID) *SignerRFC6979 {
...
+
+// NewStaticSigner creates new StaticSigner.
+func NewStaticSigner(scheme neofscrypto.Scheme, sig []byte, pubKey neofscrypto.PublicKey) *StaticSigner {
+	var id ID
+
+	switch t := pubKey.(type) {
+	case *neofsecdsa.PublicKeyRFC6979:
+		id.SetScriptHash((*keys.PublicKey)(t).GetScriptHash())
+	case *neofsecdsa.PublicKey:
+		id.SetScriptHash((*keys.PublicKey)(t).GetScriptHash())
```
Comment (roman-khimov) [inline user/signer.go:67]: Same code for both types, no need to duplicate it.

File: user/signer.go [hunk 2]
```diff
@@ -48,3 +48,40 @@ func NewSignerRFC6979WithID(pk ecdsa.PrivateKey, id ID) *SignerRFC6979 {
...
+type StaticSigner struct {
+	neofscrypto.StaticSigner
+	id ID
+}
+
+// NewStaticSigner creates new StaticSigner.
+func NewStaticSigner(scheme neofscrypto.Scheme, sig []byte, pubKey neofscrypto.PublicKey) *StaticSigner {
+	var id ID
+
+	switch t := pubKey.(type) {
```
Comment (cthulhu-rider) [inline user/signer.go:63]: `default: id = ID{}` - expected?

see also https://github.com/nspcc-dev/neofs-sdk-go/pull/479/commits/257a398926b635d38c8970a4e20c012e28d41877

**Review by cthulhu-rider — APPROVED** [review]

File: user/signer.go (hunk 1 above)
Comment (smallhive) [reply user/signer.go:67]: Agree, but in one line doesn't work.
```go
	switch t := pubKey.(type) {
	case *neofsecdsa.PublicKeyRFC6979, *neofsecdsa.PublicKey:
		id.SetScriptHash((*keys.PublicKey)(t).GetScriptHash())
	}
```
with _cannot convert t (variable of type neofscrypto.PublicKey) to type *keys.PublicKey_

File: user/signer.go (hunk 2 above)
Comment (smallhive) [reply user/signer.go:63]: no, not expected. We may throw the `panic` if type isn't good for us, for instance _PublicKeyWalletConnect_

File: user/signer.go (hunk 1 above)
Comment (smallhive) [reply user/signer.go:67]: I think this is a `type swith` limitations, like _fallthrough_ for them

File: user/signer.go (hunk 2 above)
Comment (smallhive) [reply user/signer.go:63]: In StaticSigner, unfortunately, we don't have the private key

Comment (cthulhu-rider) [reply user/signer.go:63]: lets not shrink from abstraction to particular 

we can work with `ecdsa.PublicKey` and generic constructor like in mentioned commit

File: user/signer.go (hunk 1 above)
Comment (smallhive) [reply user/signer.go]: Removed this func and the code

File: user/signer.go (hunk 2 above)
Comment (smallhive) [reply user/signer.go]: Updated

**Review by roman-khimov — APPROVED** [review]

File: user/signer.go [hunk 3]
```diff
@@ -48,3 +48,31 @@ func NewSignerRFC6979WithID(pk ecdsa.PrivateKey, id ID) *SignerRFC6979 {
...
+
+// NewStaticSignerWithID creates new StaticSigner with specifying [ID].
+func NewStaticSignerWithID(scheme neofscrypto.Scheme, sig []byte, pubKey neofscrypto.PublicKey, id ID) *StaticSigner {
+	return &StaticSigner{
+		StaticSigner: *neofscrypto.NewStaticSigner(scheme, sig, pubKey),
+		id:           id,
+	}
+}
+
+// UserID returns the [ID] using script hash calculated for the given public key.
```
Comment (cthulhu-rider) [inline user/signer.go:67]:
```suggestion
// UserID returns underlying [ID].
```

File: user/signer.go [hunk 4]
```diff
@@ -48,3 +48,31 @@ func NewSignerRFC6979WithID(pk ecdsa.PrivateKey, id ID) *SignerRFC6979 {
...
 }
+
+// StaticSigner emulates real sign and contains already precalculated hash.
+// Provides [Signer] interface.
+type StaticSigner struct {
+	neofscrypto.StaticSigner
+	id ID
+}
+
+// NewStaticSignerWithID creates new StaticSigner with specifying [ID].
```
Comment (cthulhu-rider) [inline user/signer.go:59]:
```suggestion
// NewStaticSignerWithID creates new StaticSigner with specified [ID].
```

File: user/signer.go [hunk 5]
```diff
@@ -48,3 +48,31 @@ func NewSignerRFC6979WithID(pk ecdsa.PrivateKey, id ID) *SignerRFC6979 {
...
+		id:           id,
+	}
+}
+
+// UserID returns the [ID] using script hash calculated for the given public key.
+func (s *StaticSigner) UserID() ID {
+	return s.id
+}
+
+// ResolveFromECDSAPublicKey extract [ID] like ScriptHash from corresponding key.
```
Comment (cthulhu-rider) [inline user/signer.go:72]:
```suggestion
// ResolveFromECDSAPublicKey resolves [ID] from the given [ecdsa.PublicKey].
```

**Review by cthulhu-rider — APPROVED** [review]
#400 optimization won't work for `user.StaticSigner`. May be resolved later via private interface of static signer

File: user/signer.go (hunk 3 above)
Comment (smallhive) [reply user/signer.go]: Updated

File: user/signer.go (hunk 4 above)
Comment (smallhive) [reply user/signer.go]: Updated

File: user/signer.go (hunk 5 above)
Comment (smallhive) [reply user/signer.go]: Updated

**Pull request merged**